device_doctor/lib/src/device.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'config.dart';
+import 'ios_device.dart';
+import 'utils.dart';
+
+/// Operating system on the devices that this host is configured to test.
+enum DeviceOperatingSystem { android, ios }
+
+/// Discovers available devices and chooses one to work with.
+abstract class DeviceDiscovery {
+  factory DeviceDiscovery(Config config) {
```

Comment:
We can use the flags directly in calls to this factory method.

Reply:
Replaced.